File: projectEuler/17/17.cpp
#include <iostream>
#include <string>
#include <vector>

// Function to convert a number to words
std::string numberToWords(int n) {
    std::vector<std::string> ones = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine",
                                     "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen",
                                     "seventeen", "eighteen", "nineteen"};
    std::vector<std::string> tens = {"", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};

    // Special case for 1000
    if (n == 1000)
        return "onethousand";

    std::string words;
    // Hundreds place
    if (n >= 100) {
        words += ones[n / 100] + "hundred";
        if (n % 100 != 0)
            words += "and";
    }

    // Tens and ones places
    n %= 100;
    if (n >= 20) {
        // Tens place
        words += tens[n / 10];
        // Ones place
        n %= 10;
    }
    // Ones place
    words += ones[n];

    return words;
}

// Function to count letters in the number words from 1 to 1000
int countLettersInNumbers(int limit) {
    // Count the letters in the words of numbers from 1 to limit
    int totalLetters = 0;
    for (int i = 1; i <= limit; ++i) {
        // Convert the number to words
        std::string word = numberToWords(i);
        // Count the letters in the word
        totalLetters += word.length();
    }
    return totalLetters;
}

int main() {
    int totalLetters = countLettersInNumbers(1000);
    std::cout << "Total letters used from 1 to 1000: " << totalLetters << std::endl;
    return 0;
}

File: projectEuler/17/17.py
# Function to convert a number to words
def number_to_words(n):
    # Define words for numbers 1 to 19 and multiples of ten up to 90
    ones = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine",
            "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen",
            "seventeen", "eighteen", "nineteen"]
    tens = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"]

    if n == 1000:
        return "onethousand"
    
    words = ""
    # Hundreds place
    if n >= 100:
        words += ones[n // 100] + "hundred"
        if n % 100 != 0:
            words += "and"
    
    # Tens and ones places
    n %= 100
    if n >= 20:
        words += tens[n // 10]
        n %= 10
    words += ones[n]
    
    return words

# Function to count letters in the number words from 1 to 1000
def count_letters_in_numbers(limit):
    total_letters = 0
    for i in range(1, limit + 1):
        word = number_to_words(i)
        total_letters += len(word)
    
    return total_letters

# Calculate the total letters used from 1 to 1000
total_letters = count_letters_in_numbers(1000)
print("Total letters used from 1 to 1000:", total_letters)